src/Compilers/CSharp/Test/Semantic/Semantics/MethodTypeInferenceTests.cs

```diff
@@ -633,6 +635,155 @@ static void Main(string[] args)
+        TestB(new StringOperation(), out var _);
+        TestB(new StringOperation(), out _);
+        
+        TestC<string?>(out string? discardC);
+        TestC<string?>(out string? _);
+        TestC<string?>(out var _);
+        TestC<string?>(out _);
+        
+        TestD<string?>(out string? discardD);
+        TestD<string?>(out string? _);
+        TestD<string?>(out var _);
+        TestD<string?>(out _);
+        
+        TestE(out string? discardE);
+        TestE(out var descardEVar);
+        TestE(out string? _);
+        TestE(out var _);
+        TestE(out _);
+        
+        TestF(out string? discardF);
+        TestF(out string? _);
+    }
+   
+    static void TestA<T>(IOperation<T> operation, [MaybeNull] out T result) => result = default;
+    static void TestB<T>(IOperation<T> operation, out T result) => result = default;
+    
+    static void TestC<T>([MaybeNull] out T result) => result = default;
+    static void TestD<T>(out T result) => result = default;
+
+    static void TestE([MaybeNull] out string result) => result = default;
+    static void TestF(out string result) => result = """";
+}
+";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.Net70);
+            comp.VerifyDiagnostics(new[] {
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "default").WithLocation(42, 77), // We assume that T will be non-nullable.
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "default").WithLocation(45, 52), // We assume that T will be non-nullable.
+            });
+
+            var tree = comp.SyntaxTrees.Single();
+            var model = comp.GetSemanticModel(tree);
+
+            // out _
+            ForEach(GetDiscardIdentifiers(tree), (IdentifierNameSyntax discardOut) => {
+                Assert.Null(model.GetDeclaredSymbol(discardOut));
+                var discardSymbol = (IDiscardSymbol)model.GetSymbolInfo(discardOut).Symbol;
+                Assert.Equal("System.String?", discardSymbol.Type.ToTestDisplayString());
+                Assert.Equal("System.String?", model.GetTypeInfo(discardOut).Type.ToTestDisplayString());
+            });
+
+
+            // out T _, out var _, out T? _
+            ForEach(GetDiscardDesignations(tree), (DiscardDesignationSyntax discardDecl) =>
+            {
+                Assert.Null(model.GetDeclaredSymbol(discardDecl));
+                Assert.Null(model.GetTypeInfo(discardDecl).Type);
+                Assert.Null(model.GetSymbolInfo(discardDecl).Symbol);
+                var declaration = (DeclarationExpressionSyntax)discardDecl.Parent;
+                Assert.Equal("System.String?", model.GetTypeInfo(declaration).Type.ToTestDisplayString());
+                Assert.Null(model.GetSymbolInfo(declaration).Symbol);
+            });
+        }
+
+        //Nice to have as a extension method...
+        public static IEnumerable<T> ForEach<T>(IEnumerable<T> enumeration, Action<T> action)
```

Comment:
I'm not sure I see the benefit. Simple `foreach` seems more straightforward

Reply:
I like more functional style, but I will use `foreach`.